### PR TITLE
Add legend and clean up chart

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Creates components to represent the chart legend and tooltip. Tweaks type and color styles to better match the mocks.

Screen Shot 2020-04-06 at 9 50 22 AM

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/impact-dashboard/CurveChart.tsx [hunk 1]
```diff
@@ -44,51 +55,27 @@ const ChartContainer = styled.div`
...
-
-  // TODO: factor colors out to variables somewhere?
-  // also these are not totally final
-  const colors = {
-    exposed: "#00413e",
-    infectious: "#de5558",
-    hospitalized: "#65d4ff",
-    hospitalBeds: "#de5558",
-  };
+export default function CurveChart({ curveData, hospitalBeds, markColors }) {
```
Comment (szhu) [inline src/impact-dashboard/CurveChart.tsx:58]: If possible, can you TypeScript the props?

```tsx

interface Props {
  curveData: [string, number[]][],
  hospitalBeds: number,
  markColors: MarkColors,
}

const CurveChart: React.FC<Props> = ({ curveData, hospitalBeds, markColors }) => {
   ...
}

export default CurveChart
```

This makes it easier to see how to use the component without needing to look at the implementation.

File: src/page-test/TestPage.tsx [hunk 2]
```diff
@@ -48,15 +48,15 @@ const TestPage: React.FC = () => {
       <h2>Test of news feed with data pulled from spreadsheet:</h2>
 
       {fetchedNewsFeed}
+      <h2>Test of impact dashboard</h2>
+      <div>
+        <ImpactDashboard />
+      </div>
```
Comment (szhu) [inline src/page-test/TestPage.tsx:54]: btw if you want, you can totally put this at the top of the test page! you don't have to put the news component I made first.

File: src/design-system/colors.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const colors = {
+  green: "#006C67",
+  lightBlue: "#33B6FF",
+  red: "#FF464A",
+};
+
+export default colors;
```
Comment (szhu) [inline src/design-system/colors.tsx:7]: this is nice, thank you

Comment (szhu) [reply src/design-system/colors.tsx:7]: Can we actually name constants in UpperCamelcase? thanks!

**Comment by macfarlandian** [comment]
good news, turns out that what I thought was a bug in Semiotic was actually a cryptic feature, which was demystified for me thanks to a quick response from the maintainer. So I'll probably add that to this PR before closing it since it should only require a little bit more code.

File: src/impact-dashboard/CurveChart.tsx [hunk 4]
```diff
@@ -42,81 +44,149 @@ const ChartContainer = styled.div`
...
-  // TODO: factor colors out to variables somewhere?
-  // also these are not totally final
-  const colors = {
-    exposed: "#00413e",
-    infectious: "#de5558",
-    hospitalized: "#65d4ff",
-    hospitalBeds: "#de5558",
+};
+
+interface ChartProps {
```
Comment (szhu) [inline src/impact-dashboard/CurveChart.tsx:120]: nit: Call this `Props` or `CurveChartProps`

**Pull request merged**